File: src/main/java/com/icd/survey/api/repository/survey/SurveyItemRepository.java
package com.icd.survey.api.repository.survey;

import com.icd.survey.api.entity.survey.SurveyItem;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;
import java.util.Optional;

public interface SurveyItemRepository extends JpaRepository<SurveyItem, Long> {
    Optional<List<SurveyItem>> findAllBySurveySeq(Long surveySeq);

    @Query("SELECT ao FROM SurveyItem si " +
            "JOIN si.responseList ao " +
            "WHERE si.surveySeq = :surveySeq")
    Optional<List<SurveyItem>> findAnswerOptionsBySurveySeq(@Param("surveySeq") Long surveySeq);
    Optional<List<SurveyItem>> findSurveyItemsWithAnswers(@Param("surveySeq") Long surveySeq);
}

File: survey-data/src/main/kotlin/com/ic/surveydata/answer/repository/SurveyItemRepository.kt
package com.ic.surveydata.answer.repository

import com.ic.surveydata.form.entity.SurveyFormEntity
import com.ic.surveydata.form.entity.SurveyItemEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

interface SurveyItemRepository: JpaRepository<SurveyItemEntity, String> {

    // TODO - 다시 쿼리 고민이 필요 ... !!
    @Query("""
        SELECT sf
          FROM SurveyItemEntity si
          JOIN si.surveyFormEntity sf
          JOIN si.surveyAnswers sa
          JOIN sa.surveyAnswerOptions sao
     WHERE sf.id = :surveyFormId
    """)
    fun findBySurveyFormId(@Param("surveyFormId") surveyFormId: String): List<SurveyFormEntity>

}
